cpp/src/arrow/dataset/file_base.h

```diff
@@ -34,6 +34,7 @@
 #include "arrow/filesystem/filesystem.h"
 #include "arrow/io/file.h"
 #include "arrow/util/compression.h"
+#include "arrow/util/key_value_metadata.h"
```

Comment:
This shouldn't be necessary if `arrow/type_fwd.h` is included.

Reply:
I removed this include and explicitly included `arrow/type_fwd.h`.